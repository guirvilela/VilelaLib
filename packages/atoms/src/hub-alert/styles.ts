import styled from 'styled-components';

import { HubAlertProps } from './types';

export const AlertInnerStyle = styled.div`
  display: flex;
  gap: ${({ theme }) => theme.spaces[2]};
  margin-top: -2px;
  width: 100%;
`;

export const AlertTitleStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutral[800]};
`;

export const AlertContentStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutralGrey[700]};
  font-weight: 400;
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
`;

export const AlertCloseStyle = styled.button`
  background-color: transparent;
  width: ${({ theme }) => theme.spaces[3]};
  height: ${({ theme }) => theme.spaces[3]};
  border: none;
  color: inherit;
  cursor: pointer;
`;

export const AlertStyle = styled.div<Omit<HubAlertProps, 'title'>>`
  display: flex;
  width: 100%;
  background-color: ${({ theme, color }) =>
    color ? theme.colors[color][100] : theme.colors.neutral[100]};
  color: ${({ theme, color }) =>
    color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border: 1px solid
    ${({ theme, color }) =>
      color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border-radius: ${({ theme, radius }) =>
    radius ? theme.spaces[radius] : '4px'};
  align-items: ${({ inline }) => (inline ? 'center' : 'start')};

  ${({ theme, pad }) => {
    if (typeof pad === 'string') {
      return `padding: ${theme.spaces[pad]};`;
    }
    if (Array.isArray(pad)) {
      return `padding: ${pad.reduce(
        (acc, x) => acc + `${theme.spaces[x]} `,
        ''
      )};`;
    }
    return `padding: ${theme.spaces[2]};`;
  }}

  gap: ${({ theme, gap }) => (gap ? theme.spaces[gap] : theme.spaces[2])};

  ${AlertInnerStyle} {
    flex-direction: ${({ inline }) => (inline ? 'row' : 'column')};
    align-items: ${({ inline }) => (inline ? 'center' : 'start')};
  }

  ${AlertTitleStyle} {
    font-weight: ${({ inline }) => (inline ? 400 : 600)};
    white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : undefined)};
  }
`;
