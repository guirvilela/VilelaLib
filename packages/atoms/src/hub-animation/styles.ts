import styled from 'styled-components';

import { HubAnimationProps } from './types';

export const AnimationStyle = styled.div<HubAnimationProps & { ref }>`
  ${({ theme, size, width, height }) => {
    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
          min-width: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[6]};
          height: ${theme.spaces[6]};
          min-width: ${theme.spaces[6]};
        `;
      default:
        return `
          width: ${width || theme.spaces[5]};
          height: ${height || theme.spaces[5]};
          min-width: ${width || theme.spaces[5]};
        `;
    }
  }}
`;
