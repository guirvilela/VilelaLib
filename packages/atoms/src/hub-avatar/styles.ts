import styled from 'styled-components';

import { HubAvatarProps } from './types';

export const AvatarImgStyle = styled.img`
  width: 100%;
  height: 100%;
  background-size: cover;
  border-radius: 50%;
`;

export const AvatarInitialsStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

export const AvatarStyle = styled.div<HubAvatarProps>`
  ${({ theme, size }) => {
    if (!size) {
      return `
        width: ${theme.spaces[7]};
        height: ${theme.spaces[7]};
        min-width: ${theme.spaces[7]};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[7]};
          height: ${theme.spaces[7]};
          min-width: ${theme.spaces[7]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[10]};
          height: ${theme.spaces[10]};
          min-width: ${theme.spaces[10]};
        `;
      default:
        return `
          width: ${size};
          height: ${size};
          min-width: ${size};
        `;
    }
  }}
  ${({ borderless, theme }) =>
    borderless
      ? ''
      : `
    padding: ${theme.spaces[1]};
    border: 1px solid ${theme.colors.neutral[400]};
  `};

  border-radius: 50%;

  ${AvatarInitialsStyle} {
    ${({ theme, typo }) => `
      font-family: ${theme.typography[typo || 'h4'].family};
      font-size: ${theme.typography[typo || 'h4'].size};
      line-height: ${theme.typography[typo || 'h4'].lineHeight};
      letter-spacing: ${theme.typography[typo || 'h4'].letterSpacing};
    `}

    background-color: ${({ theme, color, colorLevel }) =>
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : undefined};

    color: ${({ theme, colorText, colorTextLevel }) =>
      colorText && colorTextLevel
        ? theme.colors[colorText][colorTextLevel]
        : colorText && theme.colors[colorText]
        ? theme.colors[colorText][500]
        : colorText
        ? colorText
        : theme.colors.neutral[800]};
  }
`;
