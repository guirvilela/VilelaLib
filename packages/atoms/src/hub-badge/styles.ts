import styled, { css } from 'styled-components';

import { HubBadgeProps } from './types';

const allStyle = css<HubBadgeProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: ${({ theme }) => theme.spaces[2]};

  ${({ theme, size, width, height }) => {
    if (width || height) {
      return `
        width: ${width};
        height: ${height};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[4]};
          height: ${theme.spaces[4]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
        `;
      default:
        return '';
    }
  }}

  background-color: ${({ theme, color, colorLevel }) =>
    color && colorLevel
      ? theme.colors[color][colorLevel]
      : color
      ? theme.colors[color][100]
      : theme.colors.neutral[200]};

  color: ${({ theme, color, textColor, textColorLevel }) =>
    textColor && textColorLevel
      ? theme.colors[textColor][textColorLevel]
      : textColor
      ? theme.colors[textColor][500]
      : color
      ? theme.colors[color][500]
      : theme.colors.neutral[500]};

  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || '1']};

  ${({ theme, pad }) => {
    if (!pad) return `padding: 2px 6px`;
    if (typeof pad === 'string') return `padding: ${theme.spaces[pad]}`;
    if (pad.length === 4) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      } ${theme.spaces[pad[3]]}`;
    }
    if (pad.length === 3) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      }`;
    }
    if (pad.length === 2) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]}`;
    }
    return `padding: ${theme.spaces[pad[0]]}`;
  }};

  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  font-weight: 600;
`;

const floatStyle = css<HubBadgeProps>`
  position: relative;

  &::after {
    ${allStyle}

    position: absolute;
    content: '${({ content }) => content}';
    border: 1px solid ${({ theme }) => theme.colors.white[500]};
    ${({ theme, position }) => {
      if (position) {
        if (Array.isArray(position)) {
          return `
            top: -${theme.spaces[position[0]]};
            right: -${theme.spaces[position[1]]};
          `;
        }
        return `
          top: -${theme.spaces[position]};
          right: -${theme.spaces[position]};
        `;
      }
      return `
        top: -10px;
        right: -10px;
      `;
    }}
    ${({ theme }) => `
      min-width: ${theme.spaces[4]};
      width: ${theme.spaces[4]};
      height: ${theme.spaces[4]};
    `}
    padding: 0;

    font-family: ${({ theme }) => theme.typography.body3.family};
    font-size: ${({ theme }) => theme.typography.body3.size};
    line-height: ${({ theme }) => theme.typography.body3.lineHeight};
    letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  }
`;

export const BadgeStyle = styled.div<HubBadgeProps>`
  ${({ float, content }) => {
    if (float) {
      if (content) {
        return floatStyle;
      }
    } else {
      return allStyle;
    }
    return '';
  }}
`;
