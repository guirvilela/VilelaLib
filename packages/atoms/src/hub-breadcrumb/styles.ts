import styled from 'styled-components';

import { HubBreadcrumbProps, HubBreadcrumbsProps } from './types';

export const BreadcrumbsStyle = styled.div<HubBreadcrumbsProps>`
  display: flex;
  align-items: center;
  overflow: auto;
`;

export const BreadcrumbsPathStyle = styled.div`
  display: flex;
  align-items: center;
`;

export const BreadcrumbsIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.neutral[400]};
  margin: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const BreadcrumbStyle = styled.span<HubBreadcrumbProps>`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme, active }) =>
    active ? theme.colors.primary[500] : theme.colors.neutral[400]};
  transition: color 0.5s;
  font-weight: 600;
  white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : 'pre-wrap')};

  ${({ width }) =>
    width
      ? `
    display: block;
    max-width: ${width};
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  `
      : ''}

  &:hover {
    color: ${({ theme }) => theme.colors.primary[500]};
  }
`;
