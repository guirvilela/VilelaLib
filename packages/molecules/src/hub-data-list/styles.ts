import { HubList } from '@hubert/atoms';
import styled from 'styled-components';

export const ListStyle = styled(HubList)`
  @media print {
    max-height: unset;
    height: auto;
    overflow: unset;
  }
`;
