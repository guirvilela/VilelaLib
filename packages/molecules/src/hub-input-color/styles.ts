import styled from 'styled-components';

export const InputColor = styled.input.attrs({
  type: 'color'
})`
  background-color: transparent;
  border: none;
  width: ${({ theme }) => theme.spaces[6]};
  border-radius: ${({ theme }) => theme.spaces[2]};
`;
