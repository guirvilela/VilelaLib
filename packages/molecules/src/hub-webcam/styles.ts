import styled from 'styled-components';

export const VideoContainer = styled.video`
  @media (${({ theme }) => theme.mediaQuery.md}) {
    width: 100%;
  }
`;

export const PreviewContainer = styled.canvas`
  @media (${({ theme }) => theme.mediaQuery.md}) {
    width: 100%;
  }
`;
