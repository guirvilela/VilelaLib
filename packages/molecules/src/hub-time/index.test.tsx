import { I18nProvider } from '@hubert/i18n';
import { wrapse } from '@hubert/shared';
import { ThemeProvider } from '@hubert/styles';
import { render } from '@testing-library/react';
import React from 'react';

import { HubTime } from '.';

describe('HubTime component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <HubTime id={''} value={''} onChange={jest.fn()} />,
      {
        wrapper: wrapse(ThemeProvider, I18nProvider)
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
