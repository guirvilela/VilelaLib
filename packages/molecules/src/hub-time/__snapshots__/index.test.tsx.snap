// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubTime component matches snapshot 1`] = `
<DocumentFragment>
  .c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 8px;
  border-radius: 4px;
  border: none;
  font-weight: 600;
  -webkit-transition: background-color,outline 0.5s;
  transition: background-color,outline 0.5s;
  cursor: pointer;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
  height: auto;
  background-color: transparent;
  color: #083350;
}

.c5:disabled {
  color: #CAD3E1;
}

.c5:hover:not(:disabled) {
  color: #08273C;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  min-width: 16px;
}

.c6 svg {
  height: inherit;
  width: inherit;
  min-width: inherit;
  -webkit-transition: 0.3s ease-in-out -webkit-transform;
  -webkit-transition: 0.3s ease-in-out transform;
  transition: 0.3s ease-in-out transform;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  width: 100%;
  min-height: 40px;
  border-radius: 4px;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0 .c1 {
  border: 1px solid #A2B2BF;
  background-color: #FFFFFF;
  padding: 4px 8px;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <input
        autocomplete="off"
        class=""
        id="hub-text-field-hub-time-"
        type="text"
        value=""
      />
      <div
        class="c3"
      />
      <div
        class="c4"
        width="auto"
      >
        <button
          class="c5"
          type="button"
        >
          <div
            class="c6"
          >
            <svg
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m17 13.41-4.29-4.24a1.001 1.001 0 0 0-1.42 0l-4.24 4.24a1 1 0 1 0 1.41 1.42L12 11.29l3.54 3.54a1 1 0 0 0 1.41 0 .999.999 0 0 0 .05-1.42Z"
                fill="#083350"
              />
            </svg>
          </div>
        </button>
        <button
          class="c5"
          type="button"
        >
          <div
            class="c6"
          >
            <svg
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 9.17a1 1 0 0 0-1.41 0L12 12.71 8.46 9.17a1 1 0 1 0-1.41 1.42l4.24 4.24a.999.999 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42Z"
                fill="#083350"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
